src/cascade/input_data/emr.py

```diff
@@ -0,0 +1,219 @@
+from textwrap import dedent
+
+from scipy.interpolate import InterpolatedUnivariateSpline, LSQBivariateSpline
+
+import numpy as np
+import pandas as pd
+
+from cascade.input_data.db.mortality import get_cause_specific_mortality_data
+from cascade.input_data.db.demographics import age_groups_to_ranges, get_mean_years
+from cascade.stats import meas_bounds_to_stdev
+
+from cascade.core import getLoggers
+
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+def _prepare_prevalence(observations):
+    prevalence = observations.query("measure == 'prevalence' and sex_id in [1, 2, 3]")
+
+    zero_means = prevalence["mean"] == 0
+    if np.any(zero_means):
+        not_quite_zero = 1e-6
+        MATHLOG.debug(f"{np.sum(zero_means)} prevalence rows with 0 means. Changing them to {not_quite_zero}.")
+        prevalence.loc[zero_means, "mean"] = not_quite_zero
+
+    MATHLOG.debug(f"{len(prevalence)} prevalence observations avaliable, exluding sex_id 4")
+    return prevalence
+
+
+def _collapse_times(csmr):
+    MATHLOG.debug("Treating CSMR measurements over time ranges as measurements at the ranges midpoint")
+    csmr["time"] = (csmr["time_lower"] + csmr["time_upper"]) / 2
+    return csmr.drop(columns=["time_lower", "time_upper"])
+
+
+def _collapse_ages_unweighted(csmr):
+    csmr["age"] = (csmr["age_lower"] + csmr["age_upper"]) / 2
+    return csmr
+
+
+def _collapse_ages_weighted(execution_context, csmr):
+    mean_years = get_mean_years(execution_context)
```

Comment:
Here, I am starting to think - how is this weighting done?  the secret, I think, is in the awesome get_life_table.  So, maybe add "weighting" to the comments for get_mean_years() docstring, or maybe add a short comment to this fcn?

Reply:
Yep, this deserves a docstring.